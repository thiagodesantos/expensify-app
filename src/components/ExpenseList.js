import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Despesas</div>
            <div className="show-for-desktop">Despesa</div>
            <div className="show-for-desktop">Valor</div>
        </div>  
        <div className="list-body">
        {
            props.expenses.length === 0 ? (
              <div className="list-item list-item--message">
                <span>Não há despesas</span>
              </div>
            ) : (
                props.expenses.map((expense) => {
                    return <ExpenseListItem key={expense.id} {...expense} />
                })
            )
        }    
        </div>        
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);