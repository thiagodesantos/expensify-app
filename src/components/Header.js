import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Despesinhas</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
        <NavLink to="/criar" activeClassName="is-active">Adicionar</NavLink>
        <NavLink to="/ajuda" activeClassName="is-active">Ajuda</NavLink>
    </header>
);

export default Header;