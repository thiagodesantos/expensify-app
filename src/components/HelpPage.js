import React from 'react';

const HelpPage = () => (
    <div> Esse Ã© o help component</div>
);

export default HelpPage;