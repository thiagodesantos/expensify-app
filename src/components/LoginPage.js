import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Despesinhas</h1>
            <p>Ã‰ hora de colocar suas despesas sobre controle!</p>
            <button className="button" onClick={startLogin}>Login com o Google</button>
        </div>        
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);