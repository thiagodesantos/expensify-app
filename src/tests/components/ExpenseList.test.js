import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses'

test('Deveria renderizar ExpenseList com expeses ', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('Deveria renderizar ExpenseList sem expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});

