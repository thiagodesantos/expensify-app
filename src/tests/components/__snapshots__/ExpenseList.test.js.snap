// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deveria renderizar ExpenseList com expeses  1`] = `
<div>
  <h1>
    Lista de despesas
  </h1>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Balas"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={109500}
    createdAt={-345600000}
    description="Aluguel"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="Cartão de credito"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`Deveria renderizar ExpenseList sem expenses 1`] = `
<div>
  <h1>
    Lista de despesas
  </h1>
  <p>
    Não há despesas
  </p>
</div>
`;
